string name
string updater
---
bool removed
